flink-runtime/src/main/java/org/apache/flink/runtime/rest/handler/job/JobVertexBackPressureHandler.java

```diff
@@ -0,0 +1,134 @@
+import org.apache.flink.runtime.rest.handler.legacy.ExecutionGraphCache;
+import org.apache.flink.runtime.rest.handler.legacy.backpressure.BackPressureStatsTracker;
+import org.apache.flink.runtime.rest.handler.legacy.backpressure.OperatorBackPressureStats;
+import org.apache.flink.runtime.rest.handler.legacy.backpressure.StackTraceSampleCoordinator;
+import org.apache.flink.runtime.rest.messages.EmptyRequestBody;
+import org.apache.flink.runtime.rest.messages.JobVertexBackPressureInfo;
+import org.apache.flink.runtime.rest.messages.JobVertexIdPathParameter;
+import org.apache.flink.runtime.rest.messages.JobVertexMessageParameters;
+import org.apache.flink.runtime.rest.messages.MessageHeaders;
+import org.apache.flink.runtime.webmonitor.RestfulGateway;
+import org.apache.flink.runtime.webmonitor.retriever.GatewayRetriever;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.Executor;
+
+/**
+ * Request handler for the job vertex back pressure.
+ */
+public class JobVertexBackPressureHandler extends AbstractExecutionGraphHandler<JobVertexBackPressureInfo, JobVertexMessageParameters> {
+	/** Back pressure stats tracker. */
+	private final BackPressureStatsTracker backPressureStatsTracker;
+
+	/** Time after which stats are considered outdated. */
+	private final int refreshInterval;
+
+	public JobVertexBackPressureHandler(
+			CompletableFuture<String> localRestAddress,
+			GatewayRetriever<? extends RestfulGateway> leaderRetriever,
+			Time timeout,
+			MessageHeaders<EmptyRequestBody, JobVertexBackPressureInfo, JobVertexMessageParameters> messageHeaders,
+			ExecutionGraphCache executionGraphCache,
+			Executor executor,
+			Configuration clusterConfiguration) {
+		super(localRestAddress, leaderRetriever, timeout, messageHeaders, executionGraphCache, executor);
+
+		// Back pressure stats tracker config
+		this.refreshInterval = clusterConfiguration.getInteger(WebOptions.BACKPRESSURE_REFRESH_INTERVAL);
+		this.backPressureStatsTracker = new BackPressureStatsTracker(
+			new StackTraceSampleCoordinator(executor, 60000),
+			clusterConfiguration.getInteger(WebOptions.BACKPRESSURE_CLEANUP_INTERVAL),
+			clusterConfiguration.getInteger(WebOptions.BACKPRESSURE_NUM_SAMPLES),
+			Time.milliseconds(clusterConfiguration.getInteger(WebOptions.BACKPRESSURE_DELAY)));
+	}
+
+	@Override
+	protected JobVertexBackPressureInfo handleRequest(HandlerRequest<EmptyRequestBody, JobVertexMessageParameters> request, AccessExecutionGraph executionGraph) throws RestHandlerException {
+		JobVertexID jobVertexID = request.getPathParameter(JobVertexIdPathParameter.class);
+		AccessExecutionJobVertex accessJobVertex = executionGraph.getJobVertex(jobVertexID);
+		if (accessJobVertex instanceof ArchivedExecutionJobVertex) {
+			return new JobVertexBackPressureInfo(null, null, null, null);
+		}
+
+		ExecutionJobVertex jobVertex = (ExecutionJobVertex) accessJobVertex;
```

Comment:
This won't work in Flip-6, since we no longer send the `ExecutionJobVertex` to the rest handlers. I think in order  to make this handler work, we have to change the `BackPressureStatsTracker` and the `StackTraceSampleCoordinator`.